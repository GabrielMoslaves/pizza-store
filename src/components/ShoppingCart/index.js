import styles from "./styles.module.scss";
import { useOpener } from "../../hooks/useOpener";

const ShoppingCart = () => {
  const { setOpenCartModal } = useOpener();
  return (
    <div
      onClick={() => setOpenCartModal(true)}
      className={styles.container}
    >
      <img
        alt="carrinho"
        className={styles.img}
        src="/carrinho.png"
      />
    </div>
  );
};

export default ShoppingCart;
