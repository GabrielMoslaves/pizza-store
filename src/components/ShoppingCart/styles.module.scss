.container {
    background-color: #ffa200;
    border-radius: 50px;
    .img {
        max-width: 70px;
        cursor: pointer;
    }
    .img:hover {
        transform: scale(1.05);
    }
}

@media screen and (max-width: 655px) {
    .container {
        .img {
            max-width: 50px;
        }
    }
}
