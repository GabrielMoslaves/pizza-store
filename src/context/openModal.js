import { createContext, useState } from "react";

export const OpenModalContext = createContext();

export const OpenModalContextProvider = ({ children }) => {
  const [openCartModal, setOpenCartModal] = useState(false);
  const [openNoteModal, setOpenNoteModal] = useState(false);

  return (
    <OpenModalContext.Provider
      value={{ openCartModal, setOpenCartModal, openNoteModal, setOpenNoteModal }}
    >
      {children}
    </OpenModalContext.Provider>
  );
};
