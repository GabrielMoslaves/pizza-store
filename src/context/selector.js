import { createContext, useState } from "react";

export const SelectorContext = createContext();

export const SelectorContextProvider = ({ children }) => {
  const [selectedProducts, setSelectedProducts] = useState([]);

  return (
    <SelectorContext.Provider value={{ selectedProducts, setSelectedProducts }}>
      {children}
    </SelectorContext.Provider>
  );
};
