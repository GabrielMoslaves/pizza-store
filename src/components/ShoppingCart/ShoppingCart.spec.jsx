import { render, screen } from "@testing-library/react";
import ShoppingCart from ".";
import { OpenModalContextProvider } from "../../context/openModal";

describe("<ShoppingCart/>", () => {
  it("should render correctly", () => {
    render(
      <OpenModalContextProvider>
        <ShoppingCart />
      </OpenModalContextProvider>
    );

    expect(screen.getByRole("img", { name: /carrinho/i })).toHaveAttribute("src", "/carrinho.png");
  });

  it("should match snapshot", () => {
    const { container } = render(
      <OpenModalContextProvider>
        <ShoppingCart />
      </OpenModalContextProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
