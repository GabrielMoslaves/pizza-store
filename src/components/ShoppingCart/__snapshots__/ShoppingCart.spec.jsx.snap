// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingCart/> should match snapshot 1`] = `
<div
  class="container"
>
  <img
    alt="carrinho"
    class="img"
    src="/carrinho.png"
  />
</div>
`;
